openedx/core/djangoapps/enterprise/models.py

```diff
@@ -0,0 +1,175 @@
+    """
+    # This manager filters out some records, hence according to the Django docs it must not be used
+    # for related field access. Although False is default value, it still makes sense to set it explicitly
+    # https://docs.djangoproject.com/en/1.10/topics/db/managers/#base-managers
+    use_for_related_fields = False
+
+    def get_queryset(self):
+        """
+        Returns a new QuerySet object. Filters out inactive Enterprise Customers
+        """
+        return super(EnterpriseCustomerManager, self).get_queryset().filter(active=True)
+
+
+class EnterpriseCustomer(models.Model):
+    """
+    Enterprise Customer is an organization or a group of people that "consumes" courses: members of Enterprise
+    Customer take courses on the edX platform.
+
+    Enterprise Customer might be providing certain benefits to their members, like discounts to paid course
+    enrollments, and also might request (or require) sharing learner results with them.
+
+    Fields:
+        [implicit] id (IntegerField): EnterpriseCustomer identifier, PRIMARY KEY
+        name (CharField): Enterprise Customer name
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        sso_providers (CharField): list of comma-separated SSO provider slugs. Providers in this list should link
+                                    users registered with that provider with Enterprise Customer
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
+    """
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomer model
+        """
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_(u"Enterprise Customer name."))
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_(u"Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
+        blank=False, null=False, help_text=_(u"Data sharing policy"), max_length=20,
+        choices=(
+            (NO_DATA_SHARING, _(u"No data sharing requested")),
+            (REQUEST_DATA_SHARING, _(u"Request data sharing")),
+            (REQUIRE_DATA_SHARING, _(u"Require data sharing")),
+        )
+    )
+    sso_providers = models.CharField(
```

Comment:
Perhaps `EnterpriseCustomer` should have an immutable code/slug/short_name char field by which it could be referenced from TPA models.